### PR TITLE
docs: remove mentions of polyfills and ES5 legacy stuff

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Support for web components polyfills and ES5 transpilation has been removed in Vaadin 15.
Apparently, we missed to update some parts of the docs since then. Let's do it now.

### CODE REVIEW

File: articles/production/troubleshooting.asciidoc [hunk 1]
```diff
@@ -76,43 +76,16 @@ The bundle is located in the `META-INF/VAADIN/build/` resource folder.
...
+├── FlowBootstrap.0b77bed3.js
+├── FlowBootstrap.0b77bed3.js.br
+├── FlowClient.947c8d40.js
+├── FlowClient.947c8d40.js.br
+├── generated-flow-imports-fallback.348e3eaf.js
+├── generated-flow-imports-fallback.348e3eaf.js.br
+├── generated-flow-imports.b6dfc59f.js
+├── generated-flow-imports.b6dfc59f.js.br
+├── indexhtml.e5efd5b3.js
+└── indexhtml.e5efd5b3.js.br
```
Comment (web-padawan) [inline articles/production/troubleshooting.asciidoc:88]: These files were produced by Vaadin 23.3 when we introduced Vite support.

@mshabarov please check if this list is still accurate with Vaadin 24 🙏

File: articles/security/advanced-topics/frequent-issues.asciidoc [hunk 2]
```diff
@@ -43,13 +43,14 @@ Many tools report a CSRF vulnerability when Vaadin fetches static resources.
 These requests can't change the application state.
 Here is a list of resources that are safe to fetch without a CSRF token:
 
-- widget set files (such as `AppWidgetset.nocache.js`)
-- `vaadinBootstrap.js`
-- `vaadin-bundle-(hash).cache.js`
-- `vaadin-flow-bundle-(hash).cache.html`
-- `client-(hash).cache.js`
```
Comment (web-padawan) [inline articles/security/advanced-topics/frequent-issues.asciidoc:50]: I'm not entirely sure about removal of these, but they look like V14 leftovers.
@mshabarov Please check these out, does the new list make any sense?

**Comment by web-padawan** [comment]
Changed the PR target branch to `next`.

File: articles/production/troubleshooting.asciidoc (hunk 1 above)
Comment (mshabarov) [reply articles/production/troubleshooting.asciidoc:88]: Yes, Vaadin 24 bundle content looks exactly like this.

File: articles/security/advanced-topics/frequent-issues.asciidoc (hunk 2 above)
Comment (mshabarov) [reply articles/security/advanced-topics/frequent-issues.asciidoc:50]: Yes, it makes sense.
I'd add these entries as well:
```
web-component/web-component-(ui | bootstrap).(js | html)
VAADIN/build/webcomponenthtml-(hash).js
web-component/(exported-web-component-tag).js
```

File: articles/configuration/properties.asciidoc [hunk 3]
```diff
@@ -17,7 +17,7 @@ For Spring-based applications, there are <<{articles}/integrations/spring/config
 When using Java's system properties to set the Vaadin application parameters, the `vaadin.` prefix needs to be specified before the parameter names. The following is an example of how to set a system property when executing a Maven goal from the command-line:
 
 ----
-mvn jetty:run -Dvaadin.frontend.url.es6=http://mydomain.com/es6/ -Dvaadin.frontend.url.es5=http://mydomain.com/es5/
+mvn jetty:run -Dvaadin.pnpm.enable
```
Comment (mshabarov) [inline articles/configuration/properties.asciidoc:20]: I didn't check if it works as is, but maybe better to explicitly set `-Dvaadin.pnpm.enable=true`

Comment (web-padawan) [reply articles/configuration/properties.asciidoc]: Good point, thanks. Done.

File: articles/security/advanced-topics/frequent-issues.asciidoc (hunk 2 above)
Comment (web-padawan) [reply articles/security/advanced-topics/frequent-issues.asciidoc:50]: Thanks, added these too.

File: articles/production/troubleshooting.asciidoc (hunk 1 above)
Comment (web-padawan) [reply articles/production/troubleshooting.asciidoc]: Great, thanks. Added this file to the list.

File: articles/production/troubleshooting.asciidoc [hunk 4]
```diff
@@ -76,43 +76,17 @@ The bundle is located in the `META-INF/VAADIN/build/` resource folder.
...
+├── FlowBootstrap.0b77bed3.js.br
+├── FlowClient.947c8d40.js
+├── FlowClient.947c8d40.js.br
+├── generated-flow-imports-fallback.348e3eaf.js
+├── generated-flow-imports-fallback.348e3eaf.js.br
+├── generated-flow-imports.b6dfc59f.js
+├── generated-flow-imports.b6dfc59f.js.br
+├── indexhtml.e5efd5b3.js
+├── indexhtml.e5efd5b3.js.br
+└── webcomponenthtml-5467a14d.js
```
Comment (mshabarov) [inline articles/production/troubleshooting.asciidoc:89]:
```suggestion
├── webcomponenthtml-5467a14d.js
└── webcomponenthtml-5467a14d.js.br
```

**Review by mshabarov — APPROVED** [review]

**Pull request merged**